subscription-engine/kafka-ws/app/Redis/RedisManager.js

```diff
@@ -0,0 +1,39 @@
+const redis = require('redis');
+
+const { Logger } = require('@dojot/microservice-sdk');
+
+const logger = new Logger();
+
+const { redis: redisConfig } = require('../Config');
+
+/**
+ * Class to manage a generic purpose connection to Redis
+ */
+class RedisManager {
+  /**
+   * @constructor
+   */
+  constructor() {
+    logger.info('Creating the RedisManager singleton...');
+
+    // TODO: Implement "retry_strategy"
+    const redisOptions = {
+      host: redisConfig.host,
+      port: redisConfig.port,
+      db: redisConfig.database,
+    };
+
+    this.redisClient = redis.createClient(redisOptions);
+    logger.info('RedisManager singleton creation complete!');
+    return Object.seal(this);
+  }
+
+  /**
+   * Get the Redis connection client
+   */
+  getClient() {
+    return this.redisClient;
+  }
+}
+
+module.exports = new RedisManager();
```

Comment:
Perfect! I try to avoid singletons as much as possible, but in this case I only followed the idea of an existing implementation `RedisExpireMgmt.js`, but I did not take advantage of this one to not cause an impact. I believe that all of this will be reviewed, because the code is full of TODO everywhere.
For me, I would join the RedisExpireMgmt and RedisManager files, think about a nice structure and leave it in the SDK for other services to use, but there is no more time for that.

Reply:
Nice! So, I think we can leave a TODO here too haha what you think?

Reply:
I would take what is accessed by the routes and refactor a lot lol